src/main/kotlin/org/rust/lang/core/types/ty/TyAdt.kt

```diff
@@ -0,0 +1,50 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.lang.core.types.ty
+
+import com.intellij.codeInsight.completion.CompletionUtil
+import org.rust.lang.core.psi.ext.RsStructOrEnumItemElement
+import org.rust.lang.core.psi.ext.typeParameters
+import org.rust.lang.core.types.infer.TypeFolder
+import org.rust.lang.core.types.infer.TypeVisitor
+import org.rust.lang.core.types.type
+
+/**
+ * Represents struct/enum/union.
+ * "ADT" may be read as "Algebraic Data Type".
+ * The name is inspired by rustc
+ */
+@Suppress("DataClassPrivateConstructor")
+data class TyAdt private constructor(
+    val item: RsStructOrEnumItemElement,
+    val typeArguments: List<Ty>
```

Comment:
I think we can even use a raw Array here

Reply:
I think it won't bring any benefits